cep-20.2.md

```diff
@@ -0,0 +1,831 @@
+
+    # Allow runpath / rpath to point to these locations outside of the environment
+    # (was `runpath_whitelist`)
+    rpath_allowlist: [glob]
+
+    # what to do when detecting overdepending
+    overdepending_behavior: str # one of "ignore" or "error" (defaults to "error")
+
+    # what to do when detecting overlinking
+    overlinking_behavior: str # one of "ignore" or "error" (defaults to "error")
+
+  # Actions that are run after linking or before unlinking
+  link_options:
+    # copies fn[.bat/.sh] to the appropriate location, adds `.bat` or `.sh` to the filename
+    # script to run after linking
+    # was `post_link`
+    post_link_script: string
+    # script to run before unlinking
+    # was `pre_unlink`
+    pre_unlink_script: string
+    # message to show before linking (note: moved from `about`)
+    pre_link_message: string
+
+    # category: package or packaging?
+    # do not soft- or hard-link these files, but always copy them
+    # was `no_link`
+    always_copy_files: [glob]
+
+  # REMOVED:
+  # pre-link: string (was deprecated for a long time)
+  # Wether to include the recipe or not in the final package - should be specified on command line or other config file?
+  # include_recipe: bool (defaults to true)
+  # noarch_python: bool
+  # features: list
+  # msvc_compiler: str
+  # requires_features: dict
+  # provides_features: dict
+  # preferred_env: str
+  # preferred_env_executable_paths: list
+
+  # marked as "still experimental"
+  # pin_depends: Enum<"record" | "strict">
+  # overlinking_ignore_patterns: [glob]
+
+  # defaults to patchelf (only cudatoolkit is using `lief` for some reason)
+  # rpaths_patcher: None
+```
+
+### Script section
+
+```yaml
+script:
+  # the interpreter to use for the script
+  interpreter: str  # defaults to bash on UNIX and cmd.exe on Windows
+  # the script environment. You can use Jinja to pass through environment variables
+  # with the `env` key (`${{ env.get("MYVAR") }}`).
+  env: {str: str}
+  # secrets that are set as env variables but never shown in the logs or the environment
+  secrets: [str]
+  # The file to use as the script. Automatically adds `bat` or `sh` to the filename
+  # on Windows or UNIX respectively (if no file extension is given).
+  file: str  # build.sh  
+  # A string or list of strings that is the script contents (mutually exclusive with `file`)
+  content: str | [str]
+```
+
+## Source section
+
+```yaml
+source: [SourceElement]
+```
+
+where the different source elements are defined as follows.
+
+### URL source
+
+```yaml
+# url pointing to the source tar.gz|zip|tar.bz2|... (this can be a list of mirrors that point to the same file)
+url: url | [url]
+# destination folder in work directory
+target_directory: path
+# rename the downloaded file to this name
+file_name: str
+# hash of the file
+sha256: hex string
+# legacy md5 sum of the file (test both, prefer sha256)
+md5: hex string
+# relative path from recipe file
+patches: [path]
+```
```

Comment:
This is missing the `no_hoist` option.
(Not saying it is a very good option; but that functionality would be lost.)

Examples for where this is used are the CDT packages in conda-forge, e.g., https://github.com/conda-forge/cdt-builds/blob/66808da23b025f89cb40c9c725cf707f9caa4cfa/cdts/alsa-lib-cos7-x86_64/meta.yaml#L8

We could replace this with something better (better name, better functionality), see
https://github.com/conda/conda-build/issues/3595 ,esp., https://github.com/conda/conda-build/issues/3595#issuecomment-509339146 onward.

(An option to prevent the build tool from extracting an archive (can't remember if one exists) would also be good -- but that can be proposed later.)

Reply:
Hmm, I do not understand what no_hoist does. After reading the issue/Pr I get an idea (keep parent folders of the tar around).

I don't find it documented at all: https://docs.conda.io/projects/conda-build/en/latest/search.html?q=no_hoist

I understand that CDT packages are being phased out?

I did also think about an option to prevent the automatic extraction of a tarball. Maybe we should rather do that (and then people can do whatever they want in the build script)?

like an extra option:

```
extract: true / false
or 
auto_extract: true/false (default to true)
```

Reply:
Or `extract_archive: true / false`

Reply:
How about an "extract_archive" option @beckermr? Then in the build script we could still extract the archive with any options available to the `tar` installed in the build env.

Reply:
That will make more work for someone else when they convert recipes with it. I'm not a huge fan of that.

Reply:
No gaiiin without paiiin. More seriously, some of the reorderings here already mean that we will need a more powerful converter.

I relatively strongly disagree with the `no_hoist` option as I don't understand it from the option alone.

In fact in `rattler-build` we don't do any of the "hoisting" anyways. So I am wondering if this is the wrong idea from the start and the default should always be to "not hoist". 

If someone has a deeply nested tar ball then they will just have to cd into the right directory.

Reply:
Any decisions here? I like `extract`.